fang/src/asynk/async_queue.rs

```diff
@@ -165,165 +171,49 @@ where
+        let create_query = format!("CREATE DATABASE {} WITH TEMPLATE fang;", db_name);
+        let delete_query = format!("DROP DATABASE IF EXISTS {};", db_name);
+
+        let conn = res.pool.as_mut().unwrap().get().await.unwrap();
+        conn.execute(&delete_query, &[]).await.unwrap();
+
+        while let Err(e) = conn.execute(&create_query, &[]).await {
+            if e.as_db_error().unwrap().message()
+                != "source database \"fang\" is being accessed by other users"
+            {
+                panic!("{:?}", e);
             }
-        };
-
-        let task: Task = if !task.uniq() {
-            AsyncQueue::<NoTls>::insert_task_query(
-                transaction,
-                metadata,
-                &task.task_type(),
-                scheduled_at,
-            )
-            .await?
-        } else {
-            AsyncQueue::<NoTls>::insert_task_if_not_exist_query(
-                transaction,
-                metadata,
-                &task.task_type(),
-                scheduled_at,
-            )
-            .await?
-        };
-
-        Ok(task)
-    }
-    async fn remove_all_tasks(&mut self) -> Result<u64, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        AsyncQueue::<NoTls>::remove_all_tasks_query(transaction).await
-    }
-
-    async fn remove_all_scheduled_tasks(&mut self) -> Result<u64, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        AsyncQueue::<NoTls>::remove_all_scheduled_tasks_query(transaction).await
-    }
-
-    async fn remove_task(&mut self, id: Uuid) -> Result<u64, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        AsyncQueue::<NoTls>::remove_task_query(transaction, id).await
-    }
-
-    async fn remove_task_by_metadata(
-        &mut self,
-        task: &dyn AsyncRunnable,
-    ) -> Result<u64, AsyncQueueError> {
-        if task.uniq() {
-            let transaction = &mut self.transaction;
-
-            AsyncQueue::<NoTls>::remove_task_by_metadata_query(transaction, task).await
-        } else {
-            Err(AsyncQueueError::TaskNotUniqError)
         }
-    }
-
-    async fn remove_tasks_type(&mut self, task_type: &str) -> Result<u64, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        AsyncQueue::<NoTls>::remove_tasks_type_query(transaction, task_type).await
-    }
-
-    async fn update_task_state(
-        &mut self,
-        task: Task,
-        state: FangTaskState,
-    ) -> Result<Task, AsyncQueueError> {
-        let transaction = &mut self.transaction;
-
-        AsyncQueue::<NoTls>::update_task_state_query(transaction, task, state).await
-    }
-
-    async fn fail_task(
-        &mut self,
-        task: Task,
-        error_message: &str,
-    ) -> Result<Task, AsyncQueueError> {
-        let transaction = &mut self.transaction;
 
-        AsyncQueue::<NoTls>::fail_task_query(transaction, task, error_message).await
-    }
+        drop(conn);
```

Comment:
won't it be dropped automatically?

Reply:
nope we have tested, the alternative is to put the block inside a scope for example `{  code  }`.

Reply:
Without that `drop`, the compiler complains because `res` is returned before `conn` is dropped.